server/index.js

```diff
@@ -259,6 +259,32 @@ app.get('/api/user/listings/:listingId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.delete('/api/user/listings/:listingId', (req, res, next) => {
+  const listingId = Number(req.params.listingId);
+  if (!Number.isInteger(listingId) || listingId <= 0) {
+    throw new ClientError(400, '"listingId" must be a positive integer.');
+  }
+  const sql = `
+    delete from "listings"
+    where "listingId" = $1
+    returning *;
+    `;
```

Comment:
Here you should use the req.user.userId in your `where` clause as well, so that users can't delete each other's listings.

```sql
where "listingId" = $1
  and "userId"    = $2
```